prqlc/prql-compiler/src/sql/std.sql.prql

```diff
@@ -31,10 +31,10 @@ let average = column -> s"AVG({column:0})"
 let stddev = column -> s"STDDEV({column:0})"
 
 @{window_frame=true, coalesce="TRUE"}
-let every = column -> s"BOOL_AND({column:0})"
+let all = column -> s"BOOL_AND({column:0})"
 
 @{window_frame=true, coalesce="FALSE"}
-let any = column -> s"BOOL_OR({column:0})"
+let some = column -> s"BOOL_OR({column:0})"
```

Comment:
> some generally means "more than one"

I'd say it means "one or more".

In any case, I do agree that "any" is a better name than "some".

Reply:
... but we need a name for the "all encapsulating type". In typescript this is called `any` and in Python it is `Any`. So there is no debate here.

Reply:
For reference:

Rust has `Iterator.any`: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any

JavaScript has `Array.some`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Python has `any`: https://docs.python.org/3/library/functions.html#any

Pandas has `any`: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.any.html

Dyplr has `cumany`: https://dplyr.tidyverse.org/reference/cumall.html

Postgres has neither (or both if you count range functions): (see first note below the table) https://www.postgresql.org/docs/9.5/functions-aggregate.html

Reply:
Yeah — the root issue is that most languages can have `any` as both a function and a type, and so `any` is used for both.

How viable do you think that is for us?

Reply:
Not viable at all. I hope that the need for having `any` type in the language at all will subdue when we add generics.

Right now, I need any type for expressing `array of anything` or `tuple of anything`, but this will go away when we have generics, because you'd write `array of T` or `tuple of T`.

So maybe, for now, any type can be named `anytype` and `some` can be renamed back to `any`. And we do a rename again later, if it turns out there really is a need for any type.

Reply:
Yes, that's a great idea

Reply:
> So maybe, for now, any type can be named `anytype` and some can be renamed back to `any`. And we do a rename again later, if it turns out there really is a need for any type.

+1 on this for either `anytype` or `anything`.

Reply:
Timely topic, I learned today that polars changed horizontal `any` to `any_horizontal`.
https://github.com/eitsupi/querying-with-prql/commit/4dcdfd6b58d7860ae6d29f9299ef1175f309061e
https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.any_horizontal.html

dplyr has `if_any` for this.
https://dplyr.tidyverse.org/articles/colwise.html#filter

By the way, I think the equivalent here in dplyr is base R's `base::any`.
`dplyr::cumany` is a Window function.

``` r
any(FALSE, TRUE, FALSE)
#> [1] TRUE
dplyr::cumany(c(FALSE, TRUE, FALSE))
#> [1] FALSE  TRUE  TRUE
```

<sup>Created on 2023-10-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I also think that the `any` function is written more often then any type, so I think a type name like `anytype` makes sense.